src/app/cli/src/commands/verify_command.rs

```diff
@@ -9,23 +9,29 @@
 
 use std::sync::Arc;
 
+use futures::TryStreamExt;
 use kamu::domain::*;
 use opendatafabric::*;
 
 use super::{BatchError, CLIError, Command};
 use crate::output::OutputConfig;
 use crate::VerificationMultiProgress;
 
-type GenericVerificationResult =
-    Result<Vec<(DatasetRef, Result<VerificationResult, VerificationError>)>, CLIError>;
+type GenericVerificationResult = Result<
+    Vec<(
+        DatasetRef,
+        Arc<Result<VerificationResult, VerificationError>>,
+    )>,
+    CLIError,
+>;
 
 ///////////////////////////////////////////////////////////////////////////////
 
 pub struct VerifyCommand {
     dataset_repo: Arc<dyn DatasetRepository>,
     verification_svc: Arc<dyn VerificationService>,
     output_config: Arc<OutputConfig>,
-    refs: Vec<DatasetRef>,
+    refs: Vec<String>,
```

Comment:
We should rather have a dedicated type based on grammar, like `DatasetName`.
It will be still mostly a string inside technically, but an explicit type has a logical meaning and validation guarantee